service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformLogProcessingService.java

```diff
@@ -0,0 +1,168 @@
+    public static final String VALUE_TYPE_APPLY_COMPLETE = "apply_complete";
+    public static final String VALUE_TYPE_APPLY_ERRORED = "apply_errored";
+    public static final String KEY_MESSAGE = "@message";
+    public static final String KEY_DIAGNOSTIC_DETAIL = "diagnosticDetail";
+    public static final String KEY_DIAGNOSTIC = "diagnostic";
+    private final TerraformProperties terraformProperties;
+    private final ObjectMapper objectMapper;
+
+    public TerraformLogProcessingService(ObjectMapper objectMapper, TerraformProperties terraformProperties) {
+        this.objectMapper = objectMapper;
+        this.terraformProperties = terraformProperties;
+    }
+
+    public void saveLogToFile(CommandRequest request, List<String> logs) throws IOException {
+        Path logPath = Paths.get(terraformProperties.getWorkingDirectoryRoot()
+                + File.separator
+                + request.getContext()
+                + "-"
+                + request.getServiceId()
+                + File.separator
+                + "logs"
+        );
+
+        if (Files.notExists(logPath)) {
+            Files.createDirectories(logPath);
+        }
+
+        Path out = Files.createTempFile(logPath, System.currentTimeMillis() + "-"
+                + request.getCorrelationId() + "-job", ".log");
+
+        Files.write(out, logs, Charset.defaultCharset());
+    }
+
+    public CommandResult buildTfCommandResult(List<String> jsonLogs) {
+        if (CollectionUtils.isEmpty(jsonLogs)) {
+            throw new IllegalArgumentException("Cannot extract information empty logs");
+        }
+
+        List<Map<String, Object>> successLogs = jsonLogs.stream()
+                .map(this::parseTfOutput)
+                .filter(this::isApplyCompleteLog)
+                .map(this::simplifyApplyCompleteLog)
+                .toList();
+
+        List<Map<String, Object>> errorLogs = jsonLogs.stream()
+                .map(this::parseTfOutput)
+                .filter(this::isErrorLog)
+                .map(this::simplifyApplyErroredLog)
+                .toList();
+        JobStatus status = CollectionUtils.isEmpty(errorLogs) ? JobStatus.success : JobStatus.error;
+        return CommandResult.builder()
+                .logs(successLogs)
+                .errors(errorLogs)
+                .status(status)
+                .build();
+
+    }
+
+
+    private Map<String, Object> parseTfOutput(String json) {
+        try {
+            return objectMapper.readValue(json, Map.class);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Error parsing JSON: " + json, e);
+        }
+    }
+
+    private String extractValueFromLog(Map<String, Object> map, String key) {
+        return String.valueOf(map.get(key));
+    }
+
+    private Map<String, Object> simplifyApplyCompleteLog(Map<String, Object> expandedLogMessage) {
+        if (!isApplyCompleteLog(expandedLogMessage)) {
+            throw new UnsupportedOperationException(
```

Comment:
Seems like we lose this information on the caller:

![Screenshot 2023-11-24 at 9 10 09 AM](https://github.com/SolaceProducts/event-management-agent/assets/40175386/6ffb4d63-fc27-4092-88b6-09b495f14103)

So you can throw a generic exception in here instead.

Reply:
Yes, I'll fix it.